pkg/reconciler/pipeline/watcher.go

```diff
@@ -0,0 +1,59 @@
+package pipeline
+
+import (
+	"fmt"
+
+	appv1 "k8s.io/api/apps/v1"
+
+	dfv1 "github.com/numaproj/numaflow/pkg/apis/numaflow/v1alpha1"
+)
+
+// getVertexStatus will calculate the status of the vertices and return the status and reason
+func getVertexStatus(vertices *dfv1.VertexList) (string, string, error) {
+	for _, vertex := range vertices.Items {
+		if !vertex.Status.IsHealthy() {
+			return "False", "Progressing", nil
+		}
+	}
+	return "True", "Successful", nil
```

Comment:
Returning a string "True"/"False" is a bit confusing to me. Could you please follow the same method signature as `getDeploymentStatus` below, it's more clear.